File: solutions/3001. Minimum Moves to Capture The Queen/3001.cpp
// Time complexity: O(1)
// Space complexity: O(1)
class Solution {
public:
    int minMovesToCaptureTheQueen(int a, int b, int c, int d, int e, int f) {
        if (a == e)
            return (c == a && (b < d && d < f || b > d && d > f)) ? 2 : 1;
        if (b == f)
            return (d == f && (a < c && c < e || a > c && c > e)) ? 2 : 1;
        if (c + d == e + f)
            return (a + b == c + d && (c < a && a < e || c > a && a > e)) ? 2 : 1;
        if (c - d == e - f)
            return (a - b == c - d && (c < a && a < e || c > a && a > e)) ? 2 : 1;
        return 2;
    }
};

File: solutions/3001. Minimum Moves to Capture The Queen/3001.rs
// Time complexity: O(1)
// Space complexity: O(1)
impl Solution {
    pub fn min_moves_to_capture_the_queen(a: i32, b: i32, c: i32, d: i32, e: i32, f: i32) -> i32 {
        if a == e {
            return if c == a && ((b < d && d < f) || (b > d && d > f)) {
                2
            } else {
                1
            };
        }
        if b == f {
            return if d == f && ((a < c && c < e) || (a > c && c > e)) {
                2
            } else {
                1
            };
        }
        if c + d == e + f {
            return if a + b == c + d && ((c < a && a < e) || (c > a && a > e)) {
                2
            } else {
                1
            };
        }
        if c - d == e - f {
            return if a - b == c - d && ((c < a && a < e) || (c > a && a > e)) {
                2
            } else {
                1
            };
        }
        2
    }
}
